File: ToyLanguageInterpretor/src/model/statement/PrintStmt.java
package model.statement;

import model.Expression;
import model.ProgState;
import utils.IHeap;
import utils.IOutput;
import utils.ISymbolTable;

public class PrintStmt implements Statement{
	private Expression exp;
	
	public PrintStmt(Expression exp){
		this.exp=exp;
	}
	@Override
	public ProgState execute(ProgState p){
		ISymbolTable <String,Integer> t = p.getSymbolTable();
		IHeap<Integer,Integer> hp = p.getHeap();
		int result = exp.evaluate(t,hp);
		IOutput<Integer> out = p.getOutput();
		out.add(result);
		return null;
	}
	
	public String toString(){
		return "Print ( "+exp+")";
	}
}

File: ToyLanguageInterpretorC#/ToyLanguageC/model/PrintStmt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToyLanguageC.utils;

namespace ToyLanguageC.model
{
    class PrintStmt : Statement
    {
        Expression exp;

        public PrintStmt(Expression exp)
        {
            this.exp = exp;
        }
        public ProgState execute(ProgState state)
        {
            SymbolTable<string, int> table = state.getSymbolTable;
            Output<int> outs = state.getOutput;
            outs.add(exp.evaluate(table));
            return state;
        }

        public string toString()
        {
            return "print(" + exp.toString() + ")";
        }
    }
}
